packages/mui-material/src/styles/createTheme.ts

```diff
@@ -0,0 +1,108 @@
+import createPalette from './createPalette';
+import createThemeWithVars, {
+  CssVarsThemeOptions,
+  ColorSystem,
+  DefaultColorScheme,
+} from './createThemeWithVars';
+import createThemeNoVars, { Theme, ThemeOptions } from './createThemeNoVars';
+
+export { createMuiTheme } from './createThemeNoVars';
+export type { ThemeOptions, Theme, CssThemeVariables } from './createThemeNoVars';
+
+// eslint-disable-next-line consistent-return
+function attachColorScheme(
+  theme: { colorSchemes?: Partial<Record<string, any>> },
+  scheme: 'light' | 'dark',
+  colorScheme: boolean | Record<string, any> | undefined,
+) {
+  if (!theme.colorSchemes) {
+    return undefined;
+  }
+  if (colorScheme) {
+    theme.colorSchemes[scheme] = {
+      ...(colorScheme !== true && colorScheme),
+      palette: createPalette({
+        ...(colorScheme === true ? {} : colorScheme),
+        mode: scheme,
+      } as any), // cast type to skip module augmentation test
+    };
+  }
+}
+
+/**
+ * Generate a theme base on the options received.
+ * @param options Takes an incomplete theme object and adds the missing parts.
+ * @param args Deep merge the arguments with the about to be returned theme.
+ * @returns A complete, ready-to-use theme object.
+ */
+export default function createTheme(
+  options: Omit<ThemeOptions, 'components'> &
+    Pick<CssVarsThemeOptions, 'defaultColorScheme' | 'colorSchemes' | 'components'> & {
+      cssVariables?:
+        | boolean
+        | Pick<
+            CssVarsThemeOptions,
+            | 'colorSchemeSelector'
+            | 'disableCssColorScheme'
+            | 'cssVarPrefix'
+            | 'shouldSkipGeneratingVar'
+          >;
```

Comment:
Wouldn't it be simpler to have these options on the root instead of nested under `cssVariables`? Is there a strong reason to keep these nested?

Reply:
I think it's easier to manage and force user to provide the correct config.

Here is an example if we keep it flatten:

```js
createTheme({
  cssVariables: false,
  disableCssColorScheme: true,
  colorSchemeSelector: '...'
})
```

Users would not know at first glance if `colorSchemeSelector` and `disableCssColorScheme` are configuration of CSS variables feature. On the contrary, by keeping related config under `cssVariables` node, it force user to not make mistake because if they disable the feature, they cannot provide any config.

Reply:
In terms of implementation, it's easier because we can just spread the `cssVariables` node to the underlying function without recreating again:

```js
if (cssVariables === false) {
  // do something
}
return createThemeWithVars({ …typeof cssVariables !== 'boolean' && cssVariables });
```

But for flatten config, need to recreate the object:

```js
if (cssVariables === false) {
  // do something
}
return createThemeWithVars({
   disableCssColorScheme: options.disableCssColorScheme,
   colorSchemeSelector: options.colorSchemeSelector,
   …other variables
});
```

Reply:
Makes sense 👍🏼